x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TransportSqlQueryAction.java

```diff
@@ -126,4 +129,25 @@ private static SqlQueryResponse createResponse(SqlQueryRequest request, ZoneId z
                 header,
                 rows);
     }
+
+    @SuppressWarnings("rawtypes")
+    private static Object value(Object r, Mode mode) {
+        /*
+         * Intervals and GeoShape instances need to be serialized (as in StreamInput/Ouput serialization) as Strings
+         * since SqlQueryResponse creation doesn't have access to GeoShape nor Interval classes to make the decision
+         * so, we flatten them as Strings before being serialized.
+         * CLI gets a special treatment see {@link org.elasticsearch.xpack.sql.action.SqlQueryResponse#value()}
+         */ 
+        if (r instanceof GeoShape) {
+            return r.toString();
+        } else if (r instanceof Interval) {
+            if (mode == CLI) {
+                return r.toString();
+            } else {
+                return ((Interval) r).value();
+            }
+        }
+
+        return r;
```

Comment:
Minor nit to eliminate the number of `return`s is to assign the new value to r and return it instead:

```suggestion
        if (r instanceof GeoShape) {
            r = r.toString();
        } else if (r instanceof Interval) {
            if (mode == CLI) {
                r = r.toString();
            } else {
                r = ((Interval) r).value();
            }
        }

        return r;
```